collectors/carbonblack/collector.js

```diff
@@ -0,0 +1,160 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2020, Alert Logic, Inc
+ * @doc
+ *
+ * carbonblack class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const utils = require("./utils");
+
+
+let typeIdPaths = [];
+
+let tsPaths = [];
+
+
+class CarbonblackCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, 'carbonblack');
+    }
+
+    pawsInitCollectionState(event, callback) {
+        const startTs = process.env.paws_collection_start_ts ?
+            process.env.paws_collection_start_ts :
+            moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+        const apiNames = JSON.parse(process.env.paws_collector_param_string_1);
+        const initialStates = apiNames.map(apiName => {
+            return {
+                apiName,
+                since: startTs,
+                until: endTs,
+                nextPage: null,
+                poll_interval_sec: 1
+            }
+        });
+        return callback(null, initialStates, 1);
+
+    }
+
+    pawsGetLogs(state, callback) {
+
+        let collector = this;
+
+        let clientSecret = collector.secret;
+        if (!clientSecret) {
+            return callback("The Client Secret was not found!");
+        }
+        clientSecret = "WV6CPA8W26UCCC8W4LYZYLIA";
```

Comment:
This clientSecret has been revoked, but it still needs to be scrubbed from here regardless